lightning/src/routing/onion_message.rs

```diff
@@ -0,0 +1,283 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! [Onion message] pathfinding lives here. Finding paths for onion messages is necessary for
+//! retrieving invoices and fulfilling invoice requests from [offers]. It differs from payment
+//! pathfinding in that channel liquidity, fees, and knobs such as `htlc_maximum_msat` do not factor
+//! into path selection -- onion messages require a peer connection and nothing more. However, we
+//! still use the network graph because onion messages between channel peers are likely to be
+//! prioritized over those between non-channel peers.
+//!
+//! [Onion message]: crate::onion_message
+//! [offers]: <https://github.com/lightning/bolts/pull/798>
+
+// Note that we do not reuse `router::find_route` because it contains a lot of payment-specific
```

Comment:
nit: Not sure we need this comment anymore since you already explained it in more detail above?